src/modules/article/article.service.ts

```diff
@@ -7,75 +7,104 @@ import { MessageCodeError } from '../../shared/errors/message-code-error';
 import { UpdateArticleDto } from './dto/received/update-article.dto';
 import { User } from '../database/entities/user.entity';
 import { Category } from '../database/entities/category.entity';
+import { ToTranslit } from '../../shared/config/constants/transliterator.helper';
 
 @Injectable()
 export class ArticleService {
   constructor(
-    @InjectRepository(Article)
-    private readonly articleRepository: Repository<Article>,
-    @InjectRepository(User)
-    private readonly userRepository: Repository<User>,
-    @InjectRepository(Category)
-    private readonly categoryRepository: Repository<Category>,
+      @InjectRepository(Article)
+      private readonly articleRepository: Repository<Article>,
+      @InjectRepository(User)
+      private readonly userRepository: Repository<User>,
+      @InjectRepository(Category)
+      private readonly categoryRepository: Repository<Category>,
   ) {}
 
   async getAll(): Promise<Article[]> {
-    return this.articleRepository.find();
+    return this.articleRepository.find({ relations: ['categories'] });
   }
 
   async getById(id: string): Promise<Article> {
-    return this.articleRepository.findOne(id);
+    return this.articleRepository.findOne(id, { relations: ['categories'] });
   }
 
   async getArticlesByUserId(userId: string): Promise<Article[]> {
-    return this.articleRepository.find({
-      where: { userId },
-    });
+    return this.articleRepository.find({ where: { userId }, relations: ['categories'] });
   }
 
   async createOne(data: CreateArticleDto): Promise<Article> {
-    const existArticle = await this.articleRepository.findOne({ title: data.title });
-    if (existArticle) throw new MessageCodeError('article:exist');
+    const seoId = ToTranslit(data.title);
+    const article = await this.articleRepository.findOne({
+      where: [{ title: data.title }, { seoId }],
+    });
+    if (article) throw new MessageCodeError('article:exist');
 
-    const existUser = await this.userRepository.findOne({ id: data.userId });
-    if (!existUser) throw new MessageCodeError('user:notFound');
+    const user = await this.userRepository.findOne({ id: data.userId });
+    if (!user) throw new MessageCodeError('user:notFound');
 
-    const existCategory = await this.categoryRepository.findOne({ id: data.categoryId });
-    if (!existCategory) throw new MessageCodeError('category:notFound');
+    const categories = [];
+    let categoryId;
+    for (categoryId of data.categoriesIds) {
+      const category = await this.categoryRepository.findOne({ id: categoryId });
+      if (!category) throw new MessageCodeError('category:notFound');
+      categories.push(category);
+    }
```

Comment:
const categories = await this.categoryRepository.find({ id: data.categoriesIds });
if(categories.length!==data.categoriesIds.length) throw new MessageCodeError('category:notFound');

Reply:
done